packages/advanced-logic/src/extensions/payment-network/erc777/stream.ts

```diff
@@ -36,64 +36,171 @@ export default class Erc777StreamPaymentNetwork<
+              expectedFlowRate: extensionAction.parameters.expectedFlowRate,
+              paymentAddress: extensionAction.parameters.paymentAddress,
+              refundAddress: extensionAction.parameters.refundAddress,
+              salt: extensionAction.parameters.salt,
+            },
+            timestamp,
+          },
+        ],
+        values: {
+          ...proxyPNCreationAction.values,
+          expectedStartDate: extensionAction.parameters.expectedStartDate,
+          expectedFlowRate: extensionAction.parameters.expectedFlowRate,
+        },
+      };
     }
+
+    /* Subsequent Request Creation */
     if (
-      extensionAction.parameters.expectedFlowRate &&
-      !Utils.amount.isValid(extensionAction.parameters.expectedFlowRate)
+      extensionAction.parameters.masterRequestId &&
+      extensionAction.parameters.previousRequestId &&
+      extensionAction.parameters.recurrenceNumber
     ) {
-      throw Error('expectedFlowRate is not a valid amount');
-    }
+      if (
+        (extensionAction.parameters.masterRequestId ===
+          extensionAction.parameters.previousRequestId &&
+          extensionAction.parameters.recurrenceNumber !== 1) ||
+        (extensionAction.parameters.recurrenceNumber === 1 &&
+          extensionAction.parameters.masterRequestId !==
+            extensionAction.parameters.previousRequestId)
+      ) {
+        throw Error(
+          'recurrenceNumber must be 1 if masterRequestId and previousRequestId are equal and vice versa',
+        );
+      }
 
-    const proxyPNCreationAction = super.applyCreation(extensionAction, timestamp);
+      if (!extensionAction.version) {
+        throw Error('version is required at creation');
+      }
 
-    return {
-      ...proxyPNCreationAction,
-      events: [
-        {
-          name: 'create',
-          parameters: {
-            expectedStartDate: extensionAction.parameters.expectedStartDate,
-            expectedFlowRate: extensionAction.parameters.expectedFlowRate,
-            paymentAddress: extensionAction.parameters.paymentAddress,
-            refundAddress: extensionAction.parameters.refundAddress,
-            salt: extensionAction.parameters.salt,
+      return {
+        events: [
+          {
+            name: 'create',
+            parameters: {
+              previousRequestId: extensionAction.parameters.previousRequestId,
+              masterRequestId: extensionAction.parameters.masterRequestId,
+              recurrenceNumber: extensionAction.parameters.recurrenceNumber,
+            },
+            timestamp,
           },
-          timestamp,
+        ],
+        id: extensionAction.id,
+        type: this.extensionType,
+        values: {
+          previousRequestId: extensionAction.parameters.previousRequestId,
+          masterRequestId: extensionAction.parameters.masterRequestId,
+          recurrenceNumber: extensionAction.parameters.recurrenceNumber,
```

Comment:
There is a small possible factorization here. It would make it clear that for this kind of creation, values derive directly from parameters.